src/util-decode-mime.c

```diff
@@ -1830,11 +1830,12 @@ static int FindMimeHeader(const uint8_t *buf, uint32_t blen,
  * \param search_start The start of the search (ie. boundary=\")
  * \param search_end The end of the search (ie. \")
  * \param tlen The output length of the token (if found)
+ * \param max_len The maximum offset in which to search
  *
  * \return A pointer to the token if found, otherwise NULL if not found
  */
-static uint8_t * FindMimeHeaderToken(MimeDecField *field, const char *search_start,
-        const char *search_end, uint32_t *tlen)
+static uint8_t * FindMimeHeaderTokenRestrict(MimeDecField *field, const char *search_start,
+        const char *search_end, uint32_t *tlen, int max_len)
```

Comment:
`max_len` can be const. Also, why is it `int` and not `uint32_t`?

Reply:
It's used to compare with another value.

Reply:
AFAICS the value that we compare with is `uint32_t`, which we then cast to `int`.

Reply:
Roger; will adjust.